### PR TITLE
Update DealDetailServlet to get real restaurants & tags, and add authentication to delete

### PR DESCRIPTION
**Pull request by limli**
GET requests now returns real restaurants/tags instead of placeholders
DELETE now only works if the user is authorised, and is deleting a deal he posted.
and tests are added

### CODE REVIEW

**Review by jyx11011 — DISMISSED** [review]

File: src/main/java/com/google/step/servlets/DealDetailServlet.java [hunk 1]
```diff
@@ -27,25 +30,62 @@
...
       return;
     }
+
+    Deal deal = dealManager.readDeal(id);
+    if (deal == null) {
+      response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+
+    // user can only delete deals he created
```
Comment (SuyashLakhotia) [inline src/main/java/com/google/step/servlets/DealDetailServlet.java]:
```suggestion
    // user can only delete deals they created
```

File: src/test/java/com/google/step/servlets/DealDetailServletTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  public void testDoGet_notFound() throws IOException {
+    when(mockRequest.getPathInfo()).thenReturn(PATH_B);
+
+    servlet.doGet(mockRequest, mockResponse);
+
+    verify(mockResponse).setStatus(HttpServletResponse.SC_NOT_FOUND);
+  }
+
+  @Test
+  public void testDoDelete_sucess() throws IOException {
```
Comment (SuyashLakhotia) [inline src/test/java/com/google/step/servlets/DealDetailServletTest.java]:
```suggestion
  public void testDoDelete_success() throws IOException {
```

File: src/test/java/com/google/step/servlets/DealDetailServletTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            mockUserService);
+  }
+
+  @Test
+  public void testDoGet_success() throws IOException {
+    when(mockRequest.getPathInfo()).thenReturn(PATH_A);
+
+    servlet.doGet(mockRequest, mockResponse);
+
+    verify(mockResponse).setStatus(HttpServletResponse.SC_OK);
```
Comment (SuyashLakhotia) [inline src/test/java/com/google/step/servlets/DealDetailServletTest.java:148]: Verify response?

File: src/test/java/com/google/step/servlets/DealDetailServletTest.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  @Test
+  public void testDoGet_notFound() throws IOException {
+    when(mockRequest.getPathInfo()).thenReturn(PATH_B);
+
+    servlet.doGet(mockRequest, mockResponse);
+
+    verify(mockResponse).setStatus(HttpServletResponse.SC_NOT_FOUND);
```
Comment (SuyashLakhotia) [inline src/test/java/com/google/step/servlets/DealDetailServletTest.java:158]: Add a NOT_FOUND test for doDelete?

**Review by SuyashLakhotia — CHANGES_REQUESTED** [review]

**Review by SuyashLakhotia — APPROVED** [review]

**Pull request merged**